plugin/tags/tags.go

```diff
@@ -27,21 +28,29 @@ func New() *tags {
 // tags is the actual tags plugin that maintains a map with all
 // tags from all processed pages.
 type tags struct {
-	m map[string]*model.ListPage
+	tags      map[string]*model.ListPage
+	tagsMutex sync.RWMutex
 }
 
 // ProcessPage creates a new map entry for each tag in the processed
 // page and adds the page to the entry's list page.
 func (t *tags) ProcessPage(page *model.Page) error {
 	for _, tag := range page.Tags {
-		//sanitizing the tags like "Making Coffee" to "making-coffee"
+		// Sanitizing the tags like "Making Coffee" to "making-coffee".
 		tag.Name = strings.Replace(tag.Name, " ", "-", -1)
 		tag.Name = strings.ToLower(tag.Name)
 
-		if _, exists := t.m[tag.Name]; !exists {
+		t.tagsMutex.RLock()
+		_, tagExists := t.tags[tag.Name]
+		t.tagsMutex.RUnlock()
+
+		if !tagExists {
 			t.createListPage(tag.Name)
```

Comment:
`t.tagsMutex.RUnlock` has to be called after `t.createListPage` is executed. Otherwise, if the considererd goroutine hasn't finished `t.createListPage` yet, `tagExists` evaluates to `false` in another goroutine and causes it to call `t.createListPage` as well -> race condition